qulice-checkstyle/src/main/java/com/qulice/checkstyle/LineRanges.java

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Represent a set of LineRange objects. For example, an instance of this class
+ * could represent all the line ranges for methods in a given Java source code
+ * file.
+ *
+ * @author Jimmy Spivey (JimDeanSpivey@gmail.com)
+ * @version $Id$
+ */
+public final class LineRanges {
+
+    /**
+     * ArrayList of line ranges.
+     */
+    private final transient LineRanges.LocalCollection lines =
+        new LineRanges.LocalCollection();
+
+    /**
+     * Adds a line range to the internal collection.
+     * @param line The line range to add to the collection
+     */
+    public void add(final LineRange line) {
+        this.lines.collection().add(line);
+    }
+
+    /**
+     * Expses the iterator of the internal collection.
+     * @return Iterator pointing to the internal collections elements.
+     */
+    public Iterator<LineRange> iterate() {
+        return this.lines.collection().iterator();
+    }
+
+    /**
+     * Detects if the proposed line number is with any of the line ranges.
+     * @param line The proposed line number to check
+     * @return True if the proposed line number is within any line range.
+     */
+    public boolean inRange(final int line) {
+        return !this.lines.collection().isEmpty()
+            && FluentIterable.from(this.lines.collection())
+            .anyMatch(new LineRanges.LineWithAny(line));
+    }
+
+    /**
+     * Gets the subset of LineRanges that are within all proposed ranges. Does
+     * not return null; instead, returns empty range if there are no matches.
+     * @param ranges The ranges to filter on.
+     * @return Returns all LineRange elements that are within range.
+     */
+    public LineRanges within(final LineRanges ranges) {
+        final LineRanges result = new LineRanges();
+        final Iterator<LineRange> iterator = ranges.iterate();
+        while (iterator.hasNext()) {
+            final LineRange next = iterator.next();
+            for (final LineRange line : this.lines.collection()) {
+                if (next.within(line)) {
+                    result.add(line);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Clears the internal collection.
+     */
+    public void clear() {
+        this.lines.collection().clear();
+    }
+
+    /**
+     * Predicate to determine if a proposed line is within range of any of
+     * the line ranges.
+     */
+    private static final class LineWithAny implements Predicate<LineRange> {
+
+        /**
+         * The proposed line.
+         */
+        private final transient int proposed;
+
+        /**
+         * Default constructor.
+         * @param line The proposed line to check against all the line ranges.
+         */
+        private LineWithAny(final int line) {
+            this.proposed = line;
+        }
+
+        @Override
+        public boolean apply(final LineRange range) {
+            return range != null && range.within(this.proposed);
```

Comment:
@pinaf Without this nullcheck, findbugs complains:

```
[WARNING] FindBugs: M D NP: range must be non-null but is marked as nullable  At LineRanges.java:[line 129]
```

I think the issue here is that `Predicate` can iterate over collection's values and Java collections considers `null` an acceptable element.